### ISSUE TITLE
Consistent printing of authors and parameters in `print.epidist()`

### ISSUE THREAD

**Issue by joshwlambert** [epiverse-trace/epiparameter#92]
Ensure that the `epidist` printing method correctly prints study authors names with separator and parameters are printed to a consistent number of decimal places.

**Comment by joshwlambert** [comment]
Using `formatC()` to provide numeric formatting for `print.epidist()` (and by extension `print.vb_epidist()`). However, this is not a perfect solution. `formatC()` uses `digits` and `format` arguments to control the formatting of numerics.

When `format = NULL` (default case) it does not print to a consistent number of decimal places (even with `digits` set):

``` r
formatC(1, digits = 3, format = NULL)
#> [1] "   1"
formatC(1.2, digits = 3, format = NULL)
#> [1] " 1.2"
formatC(1.23, digits = 3, format = NULL)
#> [1] "1.23"
```

<sup>Created on 2023-03-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

`format = "f"` is good for consistent printing of decimal places, however, it has the limitations that it prints zero when numbers are small and does not use scientific notation when numbers are large:

``` r
formatC(1, digits = 3, format = NULL)
#> [1] "   1"
formatC(1, digits = 3, format = "f")
#> [1] "1.000"
formatC(1, digits = 3, format = "g")
#> [1] "   1"

formatC(1e-5, digits = 3, format = NULL)
#> [1] "1e-05"
formatC(1e-5, digits = 3, format = "f")
#> [1] "0.000"
formatC(1e-5, digits = 3, format = "g")
#> [1] "1e-05"

formatC(1e5, digits = 3, format = NULL)
#> [1] "1e+05"
formatC(1e5, digits = 3, format = "f")
#> [1] "100000.000"
formatC(1e5, digits = 3, format = "g")
#> [1] "1e+05"
```

<sup>Created on 2023-03-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Therefore, I have chosen to use `format = "f"` when numbers are within the range 9.999e-3 to 1e4 and otherwise use `format = "g"` to ensure small or large numbers are printed with scientific notation. 

One last limitation of this approach is `digits` is set to 3, however, when `format = "f"` the `digits` argument corresponds to digits after the decimal point, but when `format = "g"` the `digits` argument corresponds to significant figures (see `?formatC` for details):

``` r
formatC(1.2345, digits = 3, format = "f")
#> [1] "1.234"
formatC(1.2345, digits = 3, format = "g")
#> [1] "1.23"
```

<sup>Created on 2023-03-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I've also tried `format()` and `prettyNum()` but have found `formatC()` to be the best solution so far.

**Comment by joshwlambert** [comment]
Copying over input from @pratikunterwegs from PR #122.

```r
format_ = function(x) format(x, nsmall = 3, digits = 3, scientific = -3)

# regular formatting for smaller numbers
format_(1)
#> [1] "1.000"
format_(1.2)
#> [1] "1.200"
format_(1.234)
#> [1] "1.234"

# formatting after digits and nsmall are exceeded, rounded
format_(1.2349)
#> [1] "1.235"

# formatting for small numbers, rounded without counting zeros
format_(0.001239)
#> [1] "1.24e-03"
format(1e-5)
#> [1] "1e-05"

# formatting for large numbers
format(1e5)
#> [1] "1e+05"
format_(123456)
#> [1] "1.23e+05"
format_(123456.123456)
#> [1] "1.23e+05"
```

**Comment by joshwlambert** [comment]
Continuing discussion of formatting here instead of #122. 

@pratikunterwegs it seems the `format_()` converts to scientific notation at not very small values, e.g.

``` r
format_ = function(x) format(x, nsmall = 3, digits = 3, scientific = -3)
format_(0.1)
#> [1] "1e-01"
```

<sup>Created on 2023-03-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by joshwlambert** [comment]
This is different from the original formatting of the current implementation: 

``` r
ebola_incubation <- epiparameter::epidist(
  disease = "ebola", 
  epi_dist = "incubation", 
  prob_distribution = "gamma", 
  prob_distribution_params = c(shape = 0.1, scale = 0.1)
)
#> Citation cannot be created as either author, year or DOI is missing
ebola_incubation
#> Disease: ebola
#> Pathogen: NA
#> Epi Distribution: incubation
#> Study: No citation available
#> Distribution: gamma
#> Parameters:
#>   shape: 0.1
#>   rate: 10
```

<sup>Created on 2023-03-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by pratikunterwegs** [comment]
Fair point, I hadn't checked this for values around 0.1! Again I'm fine with your implementation, esp as this is a minor issue that shouldn't take much of your time or thought.

**Comment by joshwlambert** [comment]
Resolved in #122.
